crates/ruff_python_semantic/src/cfg/mod.rs

```diff
@@ -0,0 +1,61 @@
+pub mod graph;
+pub mod visualize;
+
+#[cfg(test)]
+mod tests {
+    use std::fmt::Write;
+    use std::fs;
+    use std::path::PathBuf;
+
+    use crate::cfg::graph::build_cfg;
+    use crate::cfg::visualize::draw_cfg;
+    use insta;
+    use ruff_python_ast::Stmt;
+    use ruff_python_parser::parse_module;
+    use ruff_text_size::Ranged;
+    use test_case::test_case;
+
+    #[test_case("no_flow.py")]
+    fn control_flow_graph(filename: &str) {
+        let path = PathBuf::from_iter(["resources/test/fixtures/cfg", filename]);
+        let source = fs::read_to_string(path).expect("failed to read file");
+        let stmts = parse_module(&source)
+            .unwrap_or_else(|err| panic!("failed to parse source: '{source}': {err}"))
+            .into_suite();
+
+        let mut output = String::new();
+
+        for (i, stmt) in stmts.into_iter().enumerate() {
+            let Stmt::FunctionDef(func) = stmt else {
+                continue;
+            };
```

Comment:
Should the test fail if the module body contains any non function statements (it might not be obvious that the framework only captures function level statements)